gulpfile.js

```diff
@@ -1,47 +1,87 @@
-const { src, dest, series, done } = require('gulp');
-const sass = require('gulp-sass');
-const rename = require('gulp-rename');
-const zip = require('gulp-zip');
-const merge = require('merge-stream');
-
-const build = () => {
-  console.log("Compiling CSS files...")
-  sass.compiler = require('node-sass');
-
-  return src('*.scss')
-    .pipe(sass({ outputStyle: 'expanded' }).on('error', sass.logError))
-    .pipe(dest('css/'));
-}
-
-const makeZip = (cyrillic) => {
-  var cssGlob, assetGlob, zipName;
-  if (cyrillic) {
-    cssGlob = 'css/*-cyrillic.css';
-    assetGlob = ['ursine/*.png', 'ursine/Cousine-Regular.woff', 'ursine/AdelleCyrillic-*.woff', 'ursine/AvenirNextCyr-*.woff']
-    zipName = 'Ursine_Cyrillic.zip';
-  } else {
-    cssGlob = ['css/*.css', '!css/*-cyrillic.css'];
-    assetGlob = ['ursine/*.png', 'ursine/Cousine-Regular.woff', 'ursine/Adelle-*.woff', 'ursine/AvenirNextLTPro-*.woff'];
-    zipName = 'Ursine.zip';
+const { src, dest, series, parallel, watch } = require("gulp");
+const sass = require("gulp-sass");
+const rename = require("gulp-rename");
+const zip = require("gulp-zip");
+const merge = require("merge-stream");
+const os = require("os");
+const path = require("path");
+
+const buildStyles = () => {
+  console.log("Compiling CSS files...");
+  sass.compiler = require("node-sass");
+
+  const regular = src(["*.scss", "!*-cyrillic.scss"])
+    .pipe(sass({ outputStyle: "expanded" }).on("error", sass.logError))
+    .pipe(dest("dist/ursine"));
+
+  const cyrillic = src("*-cyrillic.scss")
+    .pipe(sass({ outputStyle: "expanded" }).on("error", sass.logError))
+    .pipe(dest("dist/ursine-cyrillic"));
+
+  return merge(regular, cyrillic);
+};
+
+const includeAssets = () => {
+  const regular = src([
+    "ursine/*.png",
+    "ursine/Cousine-Regular.woff",
+    "ursine/Adelle-*.woff",
+    "ursine/AvenirNextLTPro-*.woff"
+  ]).pipe(dest("dist/ursine/ursine"));
+
+  const cyrillic = src([
+    "ursine/*.png",
+    "ursine/Cousine-Regular.woff",
+    "ursine/AdelleCyrillic-*.woff",
+    "ursine/AvenirNextCyr-*.woff"
+  ]).pipe(dest("dist/ursine-cyrillic/ursine"));
+
+  console.log("Including assets...");
+  return merge(regular, cyrillic);
+};
+
+const makeZip = () => {
+  const regular = src("dist/ursine/**").pipe(zip("Ursine.zip"));
+
+  const cyrillic = src("dist/ursine-cyrillic/**").pipe(
+    zip("Ursine_Cyrillic.zip")
+  );
+
+  console.log(`Building releases...`);
+  return merge(regular, cyrillic).pipe(dest("./release"));
+};
+
+const dev = () => {
+  let themeLocation;
+  switch (os.type()) {
```

Comment:
From what I can tell, themes in MacOS are stored in `~/Library/Application Support/abnerworks.Typora/themes`, where `~` is the user's home folder. Not sure how `process.env` works here, but the case for Mac would be along the lines of this:

```js
case "Darwin":
	themeLocation = "~/Library/Application Support/abnerworks.Typora/themes"
	break;
```

Not sure about Linux though

Reply:
Theme folder location in linux: `~/.config/Typora/themes`

Reply:
Theme folder location in linux: `~/.config/Typora/themes`